src/app/tour/[id]/components/TourDetails/TourDetails.tsx

```diff
@@ -0,0 +1,53 @@
+"use client";
+
+import React, { ReactElement, useState } from "react";
+import styles from "./TourDetails.module.css";
+import Overview from "@/app/tour/[id]/components/Overview/Overview";
+import Details from "@/app/tour/[id]/components/Details/Details";
+import Review from "@/app/tour/[id]/components/Review/Review";
+
+type Props = {
+  tourId: number;
+};
+
+export default function TourDetails({ tourId }: Props): ReactElement {
+  const [activeTab, setActiveTab] = useState("Overview");
+
+  const tabs = [
+    {
+      label: "اطلاعات کلی",
+      value: "Overview",
+      component: <Overview tourId={tourId} />,
```

Comment:
الان هر دفعه که Re-render اتفاق بیفته، این componentها هم از اول ساخته میشن
به جاش بهتره از تابع استفاده کنید
معمولاً اسم تابع رو render میذارن و اگه به عنوان Prop استفاده شه بهش Render Prop میگن 

```suggestion
      renderContent: (tourId: string) => <Overview tourId={tourId} />,
```